src/decision/src/LidarDecision.cpp

```diff
@@ -31,6 +32,143 @@ void LidarDecision::scanCallBack(const sensor_msgs::LaserScan::ConstPtr& raw_sca
+
+bool LidarDecision::obstacle_in_range(double min_angle, double max_angle, float obstacle_distance,
+                                      const sensor_msgs::LaserScan::ConstPtr& raw_scan) {
+    int i;
+    int init_pos_vec, end_pos_vec;
+
+    init_pos_vec = static_cast<int>((min_angle - raw_scan->angle_min) / raw_scan->angle_increment);
+    end_pos_vec = static_cast<int>((max_angle - raw_scan->angle_min) / raw_scan->angle_increment);
+
+    /*bool exist_obstacle = false;
+    for (i = init_pos_vec; i <= end_pos_vec; i++) {
+        if (raw_scan->ranges[i] < obstacle_distance) exist_obstacle = true;
+    }
+    if (exist_obstacle) return true;
+    else return false;*/
+    return std::any_of(
+             raw_scan->ranges.begin()+init_pos_vec, raw_scan->ranges.begin()+end_pos_vec,
+             [&](auto const& v){
+                 return v < obstacle_distance;
+             });
+
+}
+
+//the twist message to be sent out will be modified throught the following functions
+
+
+// v1. very crude function for turning, will not be used
+geometry_msgs::Twist LidarDecision::manage_twist_1(float obstacle_distance, float forward_speed,  float angular_speed,
+                                                 const sensor_msgs::LaserScan::ConstPtr& raw_scan){
+    geometry_msgs::Twist vel_msg;
+    bool left_block = LidarDecision::obstacle_in_range(0.0, pi/2, obstacle_distance, raw_scan);
+    bool right_block = LidarDecision::obstacle_in_range(pi/2, pi, obstacle_distance, raw_scan);
+    vel_msg.angular.x = forward_speed;
+    vel_msg.angular.y = 0;
+    vel_msg.linear.x = 0;
+    vel_msg.linear.y = 0;
+    vel_msg.linear.z = 0;
+    if (!left_block and !right_block) vel_msg.angular.z = 0;
+    if (left_block) vel_msg.angular.z = - angular_speed;
+    if (right_block) vel_msg.angular.z = angular_speed;
+    if (right_block and left_block) {
+        vel_msg.linear.x = 0;
+        vel_msg.angular.z = angular_speed;
+    }
+    return vel_msg;
+}
+
+// v2. three distance grid decisioning
+
+
+int LidarDecision::turn_left(const sensor_msgs::LaserScan::ConstPtr& raw_scan, float distance_used){
+    int left_side_value = 0;
+    int right_side_value = 0;
+    int angle_increment_steps;
+    angle_increment_steps = static_cast<int>((raw_scan->angle_max - raw_scan->angle_min) / raw_scan->angle_increment);
+
+    //ranges start at 0?
+    while ((raw_scan->ranges[left_side_value]>distance_used)||(left_side_value < angle_increment_steps/2)){
+        left_side_value++;
+    }
+    std::cout << "left_side " << left_side_value << std::endl;
+    while((raw_scan->ranges[angle_increment_steps - right_side_value] > distance_used)||(right_side_value < angle_increment_steps/2)){
+        right_side_value++;
+    }
+    std::cout << "right_side " << right_side_value << std::endl;
+    if (left_side_value > right_side_value) return 1;
+    else return (-1);
+
+}
+
+
+geometry_msgs::Twist LidarDecision::manage_twist_2(const sensor_msgs::LaserScan::ConstPtr& raw_scan) {
```

Comment:
Assuming this was named `manage_twist_2` because it is the second version of the function, it should be renamed to something like `manage_twist`. If there is another reason it's named `manage_twist_2`, should be renamed to something more specific.